src/Knp/Menu/Util/MenuManipulator.php

```diff
@@ -98,7 +98,7 @@ public function slice(ItemInterface $item, $offset, $length = null): ItemInterfa
         }
 
         $slicedItem = $item->copy();
-        $children = \array_slice($slicedItem->getChildren(), $offset, $length);
+        $children = \array_slice($slicedItem->getChildren(), (int) $offset, $length);
```

Comment:
I would not cast here. Instead, I would change the `is_numeric` to `is_int` above (line 86) and handle the case of `array_search` returning `false` with a proper exception saying the child does not exist